src/components/site-header/site-header.js

```diff
@@ -681,6 +785,117 @@ export class SiteHeader {
         this.toggleSearch(e);
       }
     }
+    if (customActionExpanded) {
+      if (
+        !this.customActionOverlay.contains(e.target) &&
+        !this.customActionLink.contains(e.target)
+      ) {
+        this.toggleCustomActionOverlay(e);
+      }
+    }
+  }
+
+  openCustomActionOverlay() {
+    if (!this.customActionOverlay || !this.customActionLink) return;
+    this.customActionOverlay.hidden = false;
+    this.customActionOverlay.setAttribute('aria-modal', 'true');
+    this.customActionLink.setAttribute('aria-expanded', 'true');
+  }
+
+  closeCustomActionOverlay() {
+    if (!this.customActionOverlay || !this.customActionLink) return;
+    this.customActionOverlay.hidden = true;
+    this.customActionOverlay.removeAttribute('aria-modal');
+    this.customActionLink.setAttribute('aria-expanded', 'false');
+  }
+
+  toggleCustomActionOverlay(e) {
+    if (!this.customActionOverlay || !this.focusTrapCustomAction) return;
+    e.preventDefault();
+    const isHidden = this.customActionOverlay.hasAttribute('hidden');
+    if (isHidden) {
+      this.openCustomActionOverlay();
+      this.focusTrapCustomAction.activate();
+      this.updateCustomActionOverlay();
+    } else {
+      this.focusTrapCustomAction.deactivate();
+    }
+  }
+
+  updateCustomActionOverlay() {
```

Comment:
would it be possible to have something more generic here? I assume that this code is more or less duplicate of the language switcher, so maybe we could have a single function for both
Note: this may be the case for some other function specific to the custom action here